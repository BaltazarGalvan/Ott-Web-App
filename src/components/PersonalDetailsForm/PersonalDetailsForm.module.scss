@use '../../styles/variables';
@use '../../styles/theme';

.title {
  margin-bottom: 24px;
  font-family: theme.$body-alt-font-family;
  font-weight: 700;
  font-size: 24px;
}

.error {
  margin-bottom: 24px;
  padding: 16px;
  color: variables.$white;
  font-family: theme.$body-font-family;
  font-size: 18px;
  background-color: theme.$form-feedback-error-bg-color;
  border-radius: 4px;
}

.continue {
  margin: variables.$base-spacing 0;
}

.form {
  max-height: 500px;
  overflow-y: auto;
  -ms-overflow-style: none; /* for Internet Explorer, Edge */
  scrollbar-width: none; /* for Firefox */
  &::-webkit-scrollbar {
    display: none; /* for Chrome, Safari, and Opera */
  }
  > div {
    margin: 12px 0;
  }
}
