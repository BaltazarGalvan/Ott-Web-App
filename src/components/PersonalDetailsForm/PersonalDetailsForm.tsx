import React from 'react';
import { useTranslation } from 'react-i18next';
import type { PersonalDetailsFormData, PersonalDetailsCustomField } from 'types/account';
import type { FormErrors } from 'types/form';

import TextField from '../TextField/TextField';
import Button from '../Button/Button';
import Dropdown from '../Dropdown/Dropdown';
import Checkbox from '../Checkbox/Checkbox';
import Radio from '../Radio/Radio';
import DateField from '../DateField/DateField';

import styles from './PersonalDetailsForm.module.scss';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  error?: string;
  errors: FormErrors<PersonalDetailsFormData>;
  values: PersonalDetailsFormData;
  submitting: boolean;
  fields: PersonalDetailsCustomField[];
};

const PersonalDetailsForm: React.FC<Props> = ({ onSubmit, onChange, values, errors, submitting, fields }: Props) => {
  const { t } = useTranslation('account');

  const renderCustomFields = () =>
    fields.map((field, index) => {
      if (field.type === 'dropdown') {
        return (
          <Dropdown
            key={index}
            value={values[field.name]}
            onChange={onChange}
            label={field.question}
            options={field.values}
            name={field.name}
            fullWidth
          />
        );
      }

      if (field.type === 'radio') {
        return <Radio name={field.name} header={field.question} key={index} values={field.values} onChange={onChange} />;
      }

      if (field.type === 'checkbox') {
        return (
          <Checkbox
            header={field.question}
            key={index}
            checked={false}
            value={field.question ?? ''}
            onChange={onChange}
            label={field.label}
            name={field.name}
          />
        );
      }

      if (field.type === 'date') {
        return (
          <DateField
            key={index}
            value={values[field.name]}
            onChange={onChange}
            label={field.label}
            placeholder={'mm/dd/yyyy'}
            error={!!errors[field.name]}
            helperText={errors[field.name]}
            name={field.name}
          />
        );
      }

      return (
        <TextField
          key={index}
          value={values[field.name]}
          onChange={onChange}
          label={field.label}
          placeholder={field.label}
          error={!!errors[field.name] || !!errors.form}
          helperText={errors[field.name]}
          name={field.name}
          type={field.type}
        />
      );
    });

  return (
    <form className={styles.form} onSubmit={onSubmit} data-testid="personal_details-form">
      <h2 className={styles.title}>{t('personal_details.title')}</h2>
      {errors.form ? <div className={styles.error}>{errors.form}</div> : null}
      {renderCustomFields()}
      <Button
        className={styles.continue}
        type="submit"
        label={t('personal_details.continue')}
        variant="contained"
        color="primary"
        size="large"
        disabled={submitting}
        fullWidth
      />
    </form>
  );
};

export default PersonalDetailsForm;
