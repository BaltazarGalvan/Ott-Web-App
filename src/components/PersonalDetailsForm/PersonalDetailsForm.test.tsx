import React from 'react';
import { render } from '@testing-library/react';

import PersonalDetailsForm from './PersonalDetailsForm';

describe('<PersonalDetailsForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <PersonalDetailsForm
        onChange={jest.fn()}
        onSubmit={jest.fn()}
        submitting={false}
        fields={[]}
        values={{
          email: '',
          password: '',
        }}
        errors={{}}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
