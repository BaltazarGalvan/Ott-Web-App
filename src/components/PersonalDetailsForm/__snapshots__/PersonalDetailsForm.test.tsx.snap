// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalDetailsForm> renders and matches snapshot 1`] = `
<div>
  <form
    class="form"
    data-testid="personal_details-form"
  >
    <h2
      class="title"
    >
      personal_details.personal_details
    </h2>
    <button
      aria-disabled="false"
      class="button continue primary fullWidth large"
      type="submit"
    >
      <span>
        personal_details.continue
      </span>
    </button>
  </form>
</div>
`;
